fdbclient/include/fdbclient/ClientStatusStats.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * ClientStatusStats.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FDBCLIENT_CLIENTSTATUSSTATS_H
+#define FDBCLIENT_CLIENTSTATUSSTATS_H
+
+#include <set>
+
+#include "fdbclient/FDBTypes.h"
+#include "fdbclient/Knobs.h"
+#include "flow/network.h"
+
+struct ClientStatusStats {
+	// The number of the examples from the client, could be greater than examples.size() as examples can be dropped if
+	// reached maximum number of examples.
+	int count;
+
+    // The examples of the status, in our situation it is the NetworkAddress/traceLogGroup pair
+	std::set<std::pair<NetworkAddress, Key>> examples;
+
+	ClientStatusStats() : count(0) {}
+};
+
+// This class allows adding examples to the ClientStatusStats object,
+struct ClientStatusStatsCollector : ClientStatusStats {
```

Comment:
This class is probably not necessary and can be consolidated into `ClientStatusStats`.

Reply:
I tending to put it into a separate class since the behavior is different. In `Status.actor.cpp`, there is no add examples behavior, but a simple collector.

Reply:
You already have `examples` defined in the base class, but not used. Generally I'd agree separating classes. In this case, adding another child class just seems not necessary, since the class is so simple.

Reply:
Discussed offline, `ClientStatusStats` allows direct access to `examples`. Here, `ClientStatusStatsCollector` should make it `private`, which only exposes `tryAddExample()` API.

Reply:
As discussed, we put the inheritance to private to hide the `examples` access.